routers/web/user/package.go

```diff
@@ -363,6 +363,27 @@ func PackageSettingsPost(ctx *context.Context) {
 			ctx.Flash.Error(ctx.Tr("packages.settings.link.error"))
 		}
 
+		ctx.Redirect(ctx.Link)
+		return
+	case "update":
+		success := func() bool {
+			description := form.Description
+			readme := form.Readme
+
+			if err := packages_model.SetDescriptions(ctx, pd.Package.ID, description, readme); err != nil {
+				log.Error("Error updating package: %v", err)
+				return false
+			}
+
+			return true
+		}()
+
+		if success {
+			ctx.Flash.Success(ctx.Tr("packages.settings.descriptions.success"))
+		} else {
+			ctx.Flash.Error(ctx.Tr("packages.settings.descriptions.error"))
+		}
```

Comment:
Why declare an extra function you only use once and that you call immediately?
```suggestion
		description := form.Description
		readme := form.Readme

		if err := packages_model.SetDescriptions(ctx, pd.Package.ID, description, readme); err != nil {
			log.Error("Error updating package: %v", err)
			ctx.Flash.Error(ctx.Tr("packages.settings.descriptions.error"))
		} else {
			ctx.Flash.Success(ctx.Tr("packages.settings.descriptions.success"))
		}
```

Reply:
https://github.com/go-gitea/gitea/blob/bb23e8dd45c69c7749900af8bf589ae102d588fd/routers/web/user/package.go#L336
The link has been updated the same way. I reused the same syntax.